src/hooks/usePeople.ts

```
@@ -0,0 +1,40 @@
+// Bloco Imports - hook usePeople
+import { useCallback, useEffect, useState } from 'react';
+import { Person } from '../types/Person';
+
+const API_URL =
+  'https://mate-academy.github.io/react_people-table/api/people.json';
+
+// Bloco usePeople - hook que busca pessoas e expõe reload
+export default function usePeople() {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchPeople = useCallback(async () => {
+    setLoading(true);
+    setError(null);
+    try {
+      const res = await fetch(API_URL);
+
+      if (!res.ok) {
+        throw new Error(`HTTP ${res.status}`);
+      }
+
+      const data: Person[] = await res.json();
+
+      setPeople(data);
+    } catch {
```

The catch block currently swallows the error (no parameter). For easier debugging, capture the error and log it (or include its message in the `error` state). Example: `} catch (err) { setPeople([]); setError('Something went wrong'); console.error(err); }`. This helps when CI/tests fail due to network/JSON issues.